cdk/api.py

```diff
@@ -0,0 +1,187 @@
+from aws_cdk import (
+    core,
+    aws_apigateway as apigw,
+    aws_cloudwatch as cloudwatch,
+    aws_iam as iam,
+    aws_logs as logs
+)
+
+from . import names
+
+class ZipApi(core.Construct):
+
+    def __init__(self, scope: core.Construct, id: str,
+            webhook_function,
+            on_demand_function,
+            ingest_allowed_ips):
+        super().__init__(scope, id)
+
+        stack_name = core.Stack.of(self).stack_name
+
+        policy = iam.PolicyDocument(
+            statements=[
+                iam.PolicyStatement(
+                    effect=iam.Effect.ALLOW,
+                    actions= ["execute-api:Invoke"],
+                    principals=[iam.AnyPrincipal()],
+                    # note that the policy is a prop of the api which cannot reference itself
+                    # see the Cloudformation documentation for api gateway policy attribute
+                    resources=[core.Fn.join('', ['execute-api:/', '*'])]
```

Comment:
still kind of agog that this really works

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-restapi.html#cfn-apigateway-restapi-policy

Reply:
yes